Formula/newrelic-cli.rb

```diff
@@ -0,0 +1,19 @@
+class NewrelicCli < Formula
+  desc 'The New Relic Command-line Interface'
+  homepage 'https://github.com/newrelic/newrelic-cli'
+  url 'https://github.com/newrelic/newrelic-cli/archive/v0.2.3.tar.gz'
+  sha256 'f09d7e1705731578da79f0d9c204a6875edbd1d78cc6e69bd9f7d00dd9919986'
+  head 'https://github.com/newrelic/newrelic-cli.git'
+
+  depends_on 'go' => :build
+
+  def install
+    ENV['PROJECT_VER'] = version
+    system 'make', 'compile-only'
+    bin.install 'bin/darwin/newrelic'
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/newrelic version 2>&1")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thanks for the feedback.  I've included a few other checks for a command that is not help and not version, as requested.  The trouble we have is that most of this tool requires an API key to be functional.  Anyway, I appreciate the review.  Let me know if that is sufficient.